### PR TITLE
Added NxSearchDropdown escape key behaviour - RSC-695 

### PR DESCRIPTION
**Pull request by entroform**
https://issues.sonatype.org/browse/RSC-695

### CODE REVIEW

File: lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx [hunk 1]
```diff
@@ -89,6 +89,16 @@ function NxSearchDropdownRender<T extends string | number = string>(
     onSearch(value.trim());
   }
 
+  const handleKeyDown: KeyboardEventHandler = event => {
+    if (event.key === 'Escape') {
```
Comment (rpokorny) [inline lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx:93]: I _think_ this is fine nowadays but I recall in the past different browsers had different ways to check for escape, so be sure this works in everything we currently support.

File: lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx [hunk 2]
```diff
@@ -89,6 +89,16 @@ function NxSearchDropdownRender<T extends string | number = string>(
...
+      onSearchTextChange('');
+
+      if (!event.defaultPrevented) {
+        event.preventDefault();
```
Comment (rpokorny) [inline lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx:97]: Why is this needed? Even if it is, is there a point in wrapping it in the conditional?

Comment (rpokorny) [reply lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx:97]: > Why is this needed? Even if it is, is there a point in wrapping it in the conditional?

I guess it would be needed if this was within a modal, to prevent esc from also closing the modal

File: lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx [hunk 3]
```diff
@@ -318,4 +318,37 @@ describe('NxSearchDropdown', function() {
...
+    expect(onSearchTextChange).not.toHaveBeenCalled();
+
+    filterInput.simulate('keyDown', keyDownEvent);
+
+    expect(escPreventDefault).toHaveBeenCalled();
+    expect(onSearchTextChange).toHaveBeenCalledWith('');
+
+    onSearchTextChange.mockClear();
+
+    dropdownMenu.simulate('keyDown', keyDownEvent);
```
Comment (rpokorny) [inline lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx:350]: Isn't the dropdown not even rendered at this point?

Comment (entroform) [reply lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx:350]: > Isn't the dropdown not even rendered at this point?

I think it's still rendered because `searchText` did not change and that's what toggles the dropdownMenu

File: lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx (hunk 1 above)
Comment (entroform) [reply lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx:93]: > I _think_ this is fine nowadays but I recall in the past different browsers had different ways to check for escape, so be sure this works in everything we currently support.

Good point. I did look at NxDropdown implementation and it also only checks for `key === 'Escape'`. Do you think that's a safe assumption to say that it works with what we currently supports 🤔 ?
https://caniuse.com/keyboardevent-key

File: lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx (hunk 2 above)
Comment (entroform) [reply lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx]: > > Why is this needed? Even if it is, is there a point in wrapping it in the conditional?
> 
> I guess it would be needed if this was within a modal, to prevent esc from also closing the modal

Yeah, the conditional is probably not needed, removed it here: 62efc9de1d1e3d19052e42ae14e214afbab3bcd4

File: lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx (hunk 1 above)
Comment (rpokorny) [reply lib/src/components/NxSearchDropdown/NxSearchDropdown.tsx:93]: > Good point. I did look at NxDropdown implementation and it also only checks for key === 'Escape'. Do you think that's a safe assumption to say that it works with what we currently supports thinking ?

Yeah following what we do in other components should be adequate

**Review by rpokorny — APPROVED** [review]
